cou/utils/juju_utils.py

```diff
@@ -710,3 +729,69 @@ async def _wait_for_active_idle() -> None:
                 raise WaitForApplicationsTimeout(msg) from error
 
         await _wait_for_active_idle()
+
+    async def wait_for_active_idle(
+        self,
+        timeout: int,
+        idle_period: int = DEFAULT_MODEL_IDLE_PERIOD,
+        apps: Optional[list[str]] = None,
+        raise_on_blocked: bool = False,
+        raise_on_error: bool = True,
+    ) -> None:
+        """Wait for application(s) to reach active idle state.
+
+        If no applications are provided, this function will wait for all COU-related applications.
+
+        :param timeout: How long (in seconds) to wait for the bundle settles before raising an
+                        WaitForApplicationsTimeout.
+        :type timeout: int
+        :param idle_period: How long (in seconds) statuses of all apps need to be `idle`. This
+                            delay is used to ensure that any pending hooks have a chance to start
+                            to avoid false positives.
+        :type idle_period: int
+        :param apps: Applications to wait, defaults to None
+        :type apps: Optional[list[str]]
+        :param raise_on_blocked: If any unit or app going into "blocked" status immediately raises
+                                 WaitForApplicationsTimeout, defaults to False.
+        :type raise_on_blocked: bool
+        :param raise_on_error: If any unit or app going into "error" status immediately raises
+                                 WaitForApplicationsTimeout, defaults to True.
+        :type raise_on_error: bool
+        """
+        await self.wait_for_idle(
+            timeout=timeout,
+            status="active",
+            idle_period=idle_period,
+            apps=apps,
+            raise_on_blocked=raise_on_blocked,
+            raise_on_error=raise_on_error,
+        )
+
+    async def resolve_all(self) -> None:
+        """Resolve all the units in the model if they are in error status."""
+        apps = await self.get_applications()
+        for app_name, app in apps.items():
+            juju_app = await self._get_application(app_name)
+            if juju_app.status != "error":
+                continue
+            for unit_name in app.units:
+                juju_unit = await self._get_unit(unit_name)
+                if juju_unit.workload_status != "error":
+                    continue
+                await juju_unit.resolved(retry=True)
+
+    async def get_application_status(self, charm_name: str) -> ApplicationStatus:
+        """Get ApplicationStatus by charm name.
+
+        :param charm_name: charm name of application
+        :type charm_name: str
+        :return: ApplicationStatus object
+        :rtype: ApplicationStatus
+        :raises ApplicationNotFound: When application is not found in the model.
+        """
+        status = await self.get_status()
```

Comment:
Why not using filter parameter in [`get_status`](https://github.com/juju/python-libjuju/blob/6c6d70d483937cec4404116c376b926106e05835/juju/model.py#L2460) function?

Reply:
The input is charm_name, the filter doesn't support charm name.

Reply:
Ok maybe we can get app name from `model` and get status only for single app?